Makefile

```
@@ -32,7 +32,8 @@ test: $(TARGET)
 	cd test/ && $(MAKE) test MAKEFLAGS=
 
 $(TARGET): $(OBJS)
-	$(CXX) $(CXXFLAGS) $(OBJS) -o $@
+	# Link fmt library after all object files, or else it will throw errors.
+	$(CXX) $(CXXFLAGS) $(OBJS) -lfmt -o $@
```

I believe you're referring to the order problem of linking to a static library[^1].
The proper way to resolve this is to add `LDLIBS` as `-lfmt` and append it to the end of the recipe:

```suggestion
	# Link fmt library after all object files, or else it will throw errors.
	$(CXX) $(CXXFLAGS) $(OBJS) -o $@ $(LDLIBS) 
```
Libraries must be linked after specifying the object files.

[^1]: Reference readings: [c++ undefined references with static library](https://stackoverflow.com/questions/2624238/c-undefined-references-with-static-library), [Why does the order in which libraries are linked sometimes cause errors in GCC?](https://stackoverflow.com/questions/45135/why-does-the-order-in-which-libraries-are-linked-sometimes-cause-errors-in-gcc)